llvm/test/lab2/sharapov_georgiy/test.ll

```diff
@@ -0,0 +1,190 @@
+; RUN: opt -load-pass-plugin=%llvmshlibdir/LoopFramer%shlibext -passes=LoopFramer -S %s | FileCheck %s
+
+define dso_local i32 @while_func() #0 {
```

Comment:
Could you also check that there is no additional loop guards inclusion if they are present there already?